### ISSUE TITLE
Multi UAV Simulation

### ISSUE THREAD

**Issue by okyspace** [Tartanrusrex/dsta-cmu-track#61]
Issue by: katcujic-cmu
 Created at: 2021-05-25 05:57:24
Closed by: amokduke
 Closed at: 2021-06-15 07:44:09

Assignees: amokduke

------------------------------------

No description provided.

**Comment by okyspace** [comment]
Commented by: amokduke
2021-06-02 03:56:59

------------------------------------

Work in progress. Requires faster machine for testing Multi-UAVs in localhost

Map sharing is working as shown in video capture:
https://drive.google.com/file/d/1GdqJsdS96r3K-q378UT3fR4PnIt6SPV7/view?usp=sharing

**Comment by okyspace** [comment]
Commented by: katcujic-cmu
2021-06-02 04:13:52

------------------------------------

No problem at all! Can you see to try the multi UAV setup on Azure?

Perhaps, the multi UAV sim on Azure can be done after the #24 , so we don't have high azure costs?

**Comment by okyspace** [comment]
Commented by: amokduke
2021-06-04 02:52:40

------------------------------------

Yeah, for the Azure runs, I will do it after #24.

One of the DSTA engineer was able to run up multi-UAVs (localhost) on his machine. We can use his runs to close this task. However he reported some issues in running up UAV 3 and UAV 4. I will check and update here.

**Comment by okyspace** [comment]
Commented by: katcujic-cmu
2021-06-10 04:59:08

------------------------------------

Ahh sorry, I missed this update. That is awesome if it can run! 

I have not checked with UAV3, UAV4 in a while (I've only checked UAV1, UAV2) so if it does not work, let me know I'll check on my side too.

**Comment by okyspace** [comment]
Commented by: katcujic-cmu
2021-06-10 16:52:34

------------------------------------

I am OK to move this task to **Done** @amokduke  if UAV1 and UAV2 are able to run and share maps. 
If UAV3 and UAV4 are not working correctly, we can crate a new task issue.

**Comment by okyspace** [comment]
Commented by: amokduke
2021-06-11 14:29:21

------------------------------------

Yes,

UAV1 and UAV2 are able to run and share maps:

![image.png](https://images.zenhubusercontent.com/60af601746d84465c9496341/bf683cb0-2675-40a0-b8c6-08e5bc49a15f)

For UAV 3 and UAV4, I do encounter some errors:

![image.png](https://images.zenhubusercontent.com/60af601746d84465c9496341/50709eeb-40df-49a6-9407-06646f3aa832)

![image.png](https://images.zenhubusercontent.com/60af601746d84465c9496341/88417e08-13bb-4434-9721-e1112c561349)

**Comment by okyspace** [comment]
Commented by: katcujic-cmu
2021-06-15 01:44:10

------------------------------------

Thank you for showing the me screenshot. I think I know what the issue is. I'll open a task to fix uav3, uav4. 

Since uav1 and uav2 are able to share (the screenshot looks good!!) I am OK to close.
- I have moved this task to DONE.

**Comment by okyspace** [comment]
Commented by: amokduke
2021-06-15 07:44:05

------------------------------------

Great!

I will move this to CLOSE then. If the task to fix uav 3/uav4 is something that DSTA can resolve, do let me know.

Thanks.
